src/commonMain/kotlin/io/github/kevincianfarini/alchemist/type/Volume.kt

```diff
@@ -107,6 +108,22 @@ public value class Volume internal constructor(private val rawCubicCentimeters:
      */
     public operator fun times(scale: Long): Volume = Volume(rawCubicCentimeters * scale)
 
+    /**
+     * Returns a volume whose value is multiplied by the specified [scale]. This operation may be rounded when the result
+     * cannot be precisely represented with a [Double] number.
+     *
+     * @throws IllegalArgumentException when this volume is [infinite][isInfinite] and [scale] is 0.0 or when this volume is 0
+     * and scale is [infinite][Double.isInfinite].
+     */
+    public operator fun times(scale: Double): Volume {
+        val intScale = scale.roundToInt()
```

Comment:
When we move this logic to attempt integer arithmetic into `SaturatingLong`, can we use `roundToLong` instead? Thinking of use cases like `1.grams * (Int.MAX_VALUE + 1.0)`.